// @aws-sdk/client-s3/dist/cjs/runtimeConfig.shared.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRuntimeConfig = void 0;
const endpoints_1 = require("./endpoints");
const url_parser_1 = require("@aws-sdk/url-parser");
/**
 * @internal
 */
const getRuntimeConfig = (config = {}) => {
    var _a, _b, _c, _d, _e, _f, _g;
    return ({
        apiVersion: "2006-03-01",
        disableHostPrefix: (_a = config.disableHostPrefix) !== null && _a !== void 0 ? _a : false,
        logger: (_b = config.logger) !== null && _b !== void 0 ? _b : {},
        regionInfoProvider: (_c = config.regionInfoProvider) !== null && _c !== void 0 ? _c : endpoints_1.defaultRegionInfoProvider,
        serviceId: (_d = config.serviceId) !== null && _d !== void 0 ? _d : "S3",
        signingEscapePath: (_e = config.signingEscapePath) !== null && _e !== void 0 ? _e : false,
        urlParser: (_f = config.urlParser) !== null && _f !== void 0 ? _f : url_parser_1.parseUrl,
        useArnRegion: (_g = config.useArnRegion) !== null && _g !== void 0 ? _g : false,
    });
};
exports.getRuntimeConfig = getRuntimeConfig;
